internal/envoy/middleware.go

```diff
@@ -103,20 +103,45 @@ func verifySPIFFE(ctx context.Context, logger hclog.Logger, store store.ReadStor
 }
 
 func parseURI(path string) (core.GatewayID, error) {
+	// TODO: would it be cleaner to switch this to RegEx parsing?
 	path = strings.TrimPrefix(path, "/")
-	tokens := strings.SplitN(path, "/", 6)
-	if len(tokens) != 6 {
-		return core.GatewayID{}, errors.New("invalid spiffe path")
-	}
-	if tokens[0] != "ns" || tokens[2] != "dc" || tokens[4] != "svc" {
-		return core.GatewayID{}, errors.New("invalid spiffe path")
+	tokens := strings.SplitN(path, "/", 8)
```

Comment:
Is there a useful link or comment that we could include explaining where the `8` comes from?

Reply:
```suggestion
        // https://pkg.go.dev/strings#SplitN
	tokens := strings.SplitN(path, "/", 8)
```

Reply:
Not documented AFAIK, but the SPIFFE path can be found in Consul source at https://github.com/hashicorp/consul/blob/40854125a5cdabe3d19a4ef77486f8eeaf0b647c/agent/connect/uri_service.go#L39-L54